services/pools.go

```diff
@@ -100,25 +100,33 @@ func (s *Service) RetrieveUSDBurnedLimit(limit uint64) ([]*models.USDBurned, err
 }
 
 func (s *Service) RetrieveDelegationUpdatedLimit(limit uint64) ([]*models.DelegationUpdated, error) {
-	iterations, last, err := s.getIterationsForLimitQuery(limit)
+	return s.RetrieveDelegationUpdated(0, 0, limit)
+}
+
+func (s *Service) RetrieveDelegationUpdated(fromBlock uint64, toBlock uint64, limit uint64) ([]*models.DelegationUpdated, error) {
+	iterations, lastBlock, err := s.getIterationsForQuery(fromBlock, toBlock, limit)
 	if err != nil {
 		return nil, err
 	}
 
 	var delegations []*models.DelegationUpdated
 
+	if fromBlock == 0 {
+		fromBlock = s.coreFirstBlock
+	}
+
 	logger.Log().WithField("layer", "Service-RetrieveDelegationUpdatedLimit").Infof(
-		"fetching DelegationUpdated with limit: %v to block: %v total iterations: %v...",
-		limit, last, iterations,
+		"fetching DelegationUpdated with limit: %v from block: %v to block: %v total iterations: %v...",
+		limit, fromBlock, lastBlock, iterations,
 	)
 
-	fromBlock := s.coreFirstBlock
-	toBlock := fromBlock + limit
+	startBlockOfIteration := fromBlock
+	endBlockOfIteration := startBlockOfIteration + limit
```

Comment:
better to check that endBlockOfIteration <= toBlock, eg. only one iteration from = 0, to = 100

Reply:
makes total sense, done